### DISCUSSION TITLE
How to pass attrs filled with context of the parent

### DISCUSSION THREAD

**Discussion by kamcio181** [jpsca/jinjax#44] [Q&A] [answered]
Hello Guys

I have a Button.jinja
```
{#def classes='' #}

<button class="{{ classes }}"
        {{ attrs.render() }}
>
    {{  content  }}
</button>
```

I want to use the component like below, so I want to pass file attribute to the parent component and use it on the child
```
{#def file #}

<Buttons.Button hx-delete="/merge/file/{{ file.id }}">
Label
</Buttons.Button>
```

Unfortunately in generated component I can see that file.id has not been replaced with the value: `hx-delete="/merge/file/{{ file.id }}"`

Is that something that is supported but I am doing it in wrong way or this is something that is not supported?

**Comment by jpsca** [comment, marked as answer]
**tl;dr:** Do this instead: `<Buttons.Button hx-delete={"/merge/file/" + file.id }>` 

Yes,  the component attributes can't work with Jinja code directly. That is because a component is converted to a macro before rendering, so what you were trying to do becomes:

```html+jinja
{% call catalog.irender("Buttons.Button", hx_delete="/merge/file/{{ file.id }}")
Label
{%- endcall %}
``` 

which is invalid.
The solution is to use something like  `<Buttons.Button hx-delete={"/merge/file/" + file.id }>` 

which is compiled to

```html+jinja
{% call catalog.irender("Buttons.Button", hx_delete="/merge/file" + file.id)
Label
{%- endcall %}
```
